### PR TITLE
Use packaged msgpack

### PR DESCRIPTION
**Pull request by Akm0d**
### What does this PR do?
Separates msgpack changes from https://github.com/saltstack/salt/pull/54878 
Adds a significant amount of testing to new msgpack module

### CODE REVIEW

**Comment by dwoz** [comment]
re-run full all

File: salt/log/handlers/fluent_mod.py [hunk 1]
```diff
@@ -455,7 +436,7 @@ def _make_packet(self, label, timestamp, data):
         packet = (tag, timestamp, data)
         if self.verbose:
             print(packet)
-        return msgpack.packb(packet)
+        return msgpack.packb(packet, _msgpack_module=msgpack)
```
Comment (dwoz) [inline salt/log/handlers/fluent_mod.py]: What is the driver allowing the use of a different msgpack module. I think we should keep things consistent by always using the module imported in `utils.msgpack`

**Review by dwoz — CHANGES_REQUESTED** [review]

**Review by waynew — APPROVED** [review]
🚀

**Comment by Akm0d** [comment]
re-run full archlts

**Comment by Akm0d** [comment]
re-run full all

**Comment by Akm0d** [comment]
re-run full tcp

**Comment by Akm0d** [comment]
re-run full amazon1

**Comment by Akm0d** [comment]
re-run full debian8

**Comment by Akm0d** [comment]
Cleaned up the history now that it's all green :+1:. @dwoz I tried to move as much msgpack logic as possible from salt/payload.py to salt/utils/msgpack.py.   If I move much more then the msgpack module starts behaving too differently from the base msgpack module

**Comment by Akm0d** [comment]
re-run full amazon1

**Comment by Akm0d** [comment]
re-run full pycryptodomex

**Comment by Akm0d** [comment]
re-run full fedora30

**Comment by Akm0d** [comment]
re-run full debian8

**Review by waynew — APPROVED** [review]
Does this replace all of the `import msgpack`?

It would be full of winning if we could add a lint rule to stop any `import msgpack` - though I'm not sure if those rules are in this repo.

**Comment by Akm0d** [comment]
re-run full windows2016

**Comment by Akm0d** [comment]
re-run full windows2019

**Comment by Akm0d** [comment]
> Does this replace all of the `import msgpack`?
> 
> It would be full of winning if we could add a lint rule to stop any `import msgpack` - though I'm not sure if those rules are in this repo.

It replaces most `import msgpack`.  The exception is a few linters, loggers, and tests where it would cause a circular dependency

**Comment by Akm0d** [comment]
re-run full debian8

**Comment by Akm0d** [comment]
re-run full macosxmojave

**Review by dwoz — APPROVED** [review]

**Pull request merged**